rosbag2/src/rosbag2/sequential_reader.cpp

```diff
@@ -71,9 +80,29 @@ SequentialReader::open(
   }
 }
 
+void SequentialReader::get_next_file()
+{
+  current_file_ = file_paths_.front();
+  file_paths_.erase(file_paths_.begin());
```

Comment:
Dropping information there seems unnecessary, why not just replacing current_file_ by current_file_index_?

Also, the reason why std::vector does not have pop_front() is because it is inefficient, std::dequeue is probably closer to what we'd like if we really want to erase here.